frontend/src/bundles/talent-onboarding/store/actions.ts

```diff
@@ -9,8 +9,32 @@ const updateTalentDetails = createAsyncThunk<
     UserDetailsGeneralCustom,
     UserDetailsGeneralCustom,
     AsyncThunkConfig
->(`${sliceName}/update-talent-details`, (registerPayload) => {
-    return registerPayload;
+>(
+    `${sliceName}/update-talent-details`,
+    (registerPayload, { getState, extra }) => {
+        const { talentOnBoardingApi } = extra;
+
+        if ('badges' in registerPayload) {
+            return registerPayload;
+        }
+        const { talentOnBoarding } = getState();
+
+        return talentOnBoarding.completedStep
+            ? talentOnBoardingApi.updateUserDetails(registerPayload)
+            : talentOnBoardingApi.createUserDetails(registerPayload);
```

Comment:
create 3 action

1. POST
2. PATCH
3. combine logic where you should dispatch POST or PATCH